heron/api/src/java/com/twitter/heron/api/Config.java

```diff
@@ -452,6 +487,28 @@ public static void setTopologyExactlyOnceEnabled(Map<String, Object> conf, boole
     conf.put(Config.TOPOLOGY_EXACTLYONCE_ENABLED, String.valueOf(exactOnce));
   }
 
+  public static void setTopologyPackingAlgorithm(Map<String, Object> conf,
+                                                 PackingAlgorithmType type) {
```

Comment:
Thanks @billonahill I agree with the interpretation of the `api` and `spi` packages. 

I think packing configuration is rarely provided by topology developer. It is rather commonly specified by the administrator or the member deploying the topology via `yaml` config files or command line arguments. To me this fits the definition of system-level API. WDYT?